task/buildah/0.1/buildah.yaml

```diff
@@ -107,29 +107,36 @@ spec:
       sed -i 's/^short-name-mode = .*/short-name-mode = "disabled"/' /etc/containers/registries.conf
 
       if [ "${HERMETIC}" == "true" ]; then
-        NETWORK="--network none"
+        BUILDAH_ARGS="--pull=never"
+        UNSHARE_ARGS="--net"
+        for image in $(grep -i '^FROM' "$dockerfile_path" | awk '{print $2}'); do
+          buildah pull $image
+        done
         echo "Build will be executed with network isolation"
       fi
 
-      if [ ! -z "${PREFETCH_INPUT}" ]; then
+      if [ -n "${PREFETCH_INPUT}" ]; then
         VOLUME_MOUNTS="
           --volume $(realpath ./cachi2/output):/cachi2/output:U \
           --volume $(realpath ./cachi2/cachi2.env):/cachi2/cachi2.env:U"
-
+        sed -i 's|^\s*run |RUN source /cachi2/cachi2.env \&\& \\\n    |i' "$dockerfile_path"
         echo "Prefetched content will be made available"
       fi
 
-      buildah bud \
+      # Creating new namespace to run buildah
+      echo 'root:1:4294967294' | tee -a /etc/subuid >> /etc/subgid
+      unshare -Ufp $UNSHARE_ARGS --keep-caps -r --map-users 1,1,65536 --map-groups 1,1,65536 -- buildah bud \
```

Comment:
What are the numbers based on? I went through the discussion in STONEBLD-660, but it's still not clear to me

4294967294 is some magic number?

65536 is the "number of UIDs and GIDs available"? How do we know how many are available?

Reply:
@nalind you might have some more context for the magic number?

Reply:
Ah nevermind, it's the maximum possible number isn't it
```python
In [3]: 2**32 - 2
Out[3]: 4294967294
```

Reply:
UIDs and GIDs are currently 32-bit unsigned numbers.  42949672945, as you've noted, is more or less what you see when you cast -1 to an unsigned 32-bit number, and the kernel uses that as a magic "invalid UID/GID" value, so the kernel won't allow it to be included in mappings.
65536 is what we usually allocate in /etc/subuid and /etc/subgid when a user account is created with `useradd`, so that's an educated guess.  If we're already UID 0 and we don't actually want to change the current UID/GID mappings, a more robust implementation could look at /proc/self/uid_map and /proc/self/gid_map to see which UIDs and GIDs are mapped in the parent user namespace, and use the values found there to map the ranges it finds there onto themselves.

Reply:
Ack, thanks for the explanation :+1: